nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -246,6 +261,29 @@ void BtActionServer<ActionT>::executeCallback()
   }
 }
 
+template<class ActionT>
+void BtActionServer<ActionT>::populateErrorCode(
+  typename std::shared_ptr<typename ActionT::Result> result)
+{
+  int highest_priority_error_code = 0;
+  for (const auto & error_code_id_name : error_code_id_names_) {
+    int current_error_code = 0;
+    try {
+      current_error_code = blackboard_->get<int>(error_code_id_name);
+    } catch (...) {
+      RCLCPP_ERROR(
+        logger_,
+        "Failed to get %s from blackboard",
+        error_code_id_name.c_str());
+    }
+    if (current_error_code > highest_priority_error_code) {
```

Comment:
Can you update nav2_system_test test(s) that are testing failure cases to check that this error code is non-NONE?  If we know exactly which error code should trigger for a failure, check that specific code.

Reply:
I am not quite sure which tests you are referring to?

Reply:
In nav2_system_tests, there are tests for WPF / Nav2Pose / NavThroughPoses / etc. Some of those tests specifically are designed to fail (e.g. giving it bogus goals or starting off the map, etc) so after this PR, they should have error codes that you can read in the test and make sure not only did they fail in the action response, but that the error code is populated and correct.

Reply:
So, 

inside the nav2 system tests the way point follower has one test that causes global planning to fail.  

In the _planning_ dir random path are being generated so there isn't any error codes we can check. 

I don't really see anywhere else in nav2_system_test that actually fail the planner and controller.

Reply:
I don't understand, the WPF test uses the full stack and has a failure mode of an off-map request, that should be pretty clear-cut. There's also the system_failure tests too to choose from. For the system-level, it doesn't need to be both planner/controller (later unit tests for the planner/controller server that we've discussed will do that), it needs to just be a test to evaluate how once any of the servers fail, its being properly communicated back to the BT navigator or waypoint follower or what have you. The WPF is, to me, the best choice, because that will exercise not only the BT Nav stuff, but also the WPF stuff simultaneously 

We need some kind of coverage for this feature and those seem like really easy / obvious places for it.

Reply:
Okay, 

I wasn't quite sure _what_ we wanted to test. This could be added to the error_code test and would be quicker (don't have to run through recoveries on failure by running with a similar BT). Once we update the main behavior trees to handle the errors this should be a lot faster and we could update the follow waypoint test.